backend/projects/views/dimension.py

```diff
@@ -0,0 +1,58 @@
+from rest_framework import generics
+from rest_framework.permissions import IsAuthenticated
+
+from projects.models import DynamicDimension, DimensionMetaData, ProjectDimension
+from projects.permissions import IsProjectAdmin, IsProjectStaffAndReadOnly
+from projects.serializers import DynamicDimensionSerializer, DimensionMetaDataSerializer, ProjectDimensionSerializer
+from label_types.models import ScaleType
+
+
+class DimensionMetaDataList(generics.ListCreateAPIView):
+    serializer_class = DimensionMetaDataSerializer
+    pagination_class = None
+    permission_classes = [IsAuthenticated & (IsProjectAdmin | IsProjectStaffAndReadOnly)]
+
+    def get_queryset(self):
+        return DimensionMetaData.objects.filter(dimension=self.kwargs["dimension_id"])
+
+
+class DynamicDimensionList(generics.ListCreateAPIView):
+    serializer_class = DynamicDimensionSerializer
+    pagination_class = None
+    permission_classes = [IsAuthenticated & (IsProjectAdmin | IsProjectStaffAndReadOnly)]
+
+    def get_queryset(self):
+        return DynamicDimension.objects.all()
+
+    def perform_create(self, serializer):
+        serializer.save()
+
+
+class GetAllDynamicDimensions(generics.ListAPIView):
+    serializer_class = DynamicDimensionSerializer
+    pagination_class = None
+    permission_classes = [IsAuthenticated]
+
+    def get_queryset(self):
+        return DynamicDimension.objects.all()
+
+
+class AddProjectDimension(generics.CreateAPIView):
+    serializer_class = ProjectDimensionSerializer
+
```

Comment:
Sorry, there is a minor change required here 😅. For some weird reason, when I do manual testing, this view works perfectly just fine. However, when using a test script, it throws a 403 error. The reason is apparently missing `permission_classes`. I had to add the following line to make it pass the test script:
```
permission_classes = [IsAuthenticated & (IsProjectAdmin | IsProjectStaffAndReadOnly)]
```

So I hope you can add this line before merging the PR 😅

Reply:
Yeah thank you XD probably I missed that permission. Added :D